src/Symfony/Component/Ldap/Adapter/ExtLdap/Query.php

```diff
@@ -18,6 +18,8 @@
 /**
  * @author Charles Sarrazin <charles@sarraz.in>
  * @author Bob van de Vijver <bobvandevijver@hotmail.com>
+ *
+ * @deprecated Use PaginableQuery instead if using PHP 7.3 or greater
```

Comment:
Is it really necessary to deprecate this class and create a new one? To me, the new code looks like it could be merged into the existing class.

Reply:
Unfortunately it is not that easy. In a first [attempt](https://github.com/symfony/symfony/pull/38392/commits/357ddcb03871f29c3183186a16c4bc3db6a014a9) I've tried this solution but, in order to let the two way to handle pagination to cohesist, I was using the [serverctrls](https://github.com/symfony/symfony/pull/38392/commits/357ddcb03871f29c3183186a16c4bc3db6a014a9#diff-c0e22b9cfa83e41d100fd47a035b44adR250) parameter. But that parameter, which is the only alternative to clear the pagination, is only available in PHP 7.3 (https://www.php.net/manual/en/function.ldap-parse-result.php)